### PR TITLE
tree-wide: introduce hashmap_ensure_put() and use it

### PR DESCRIPTION
**Pull request by ssahani**

### CODE REVIEW

File: src/network/wait-online/wait-online.c [hunk 1]
```diff
@@ -83,14 +83,12 @@ static int parse_interface_with_operstate_range(const char *str) {
...
 
-        r = hashmap_ensure_allocated(&arg_interfaces, &string_hash_ops);
-        if (r < 0)
+        r = hashmap_ensure_put(&arg_interfaces, &string_hash_ops, ifname, TAKE_PTR(range));
+        if (r == -ENOMEM)
                 return log_oom();
-
-        r = hashmap_put(arg_interfaces, ifname, TAKE_PTR(range));
-        if (r < 0)
+        else if (r < 0)
```
Comment (bluca) [inline src/network/wait-online/wait-online.c]: ```else``` are not really needed, given it returns? just a nit

**Review by bluca — APPROVED** [review]
LGTM

File: src/network/wait-online/wait-online.c (hunk 1 above)
Comment (ssahani) [reply src/network/wait-online/wait-online.c]: Updated thanks for the review.

File: src/network/networkd-network.c [hunk 2]
```diff
@@ -142,12 +142,10 @@ static int network_resolve_stacked_netdevs(Network *network) {
...
 
-                r = hashmap_ensure_allocated(&network->stacked_netdevs, &string_hash_ops);
-                if (r < 0)
+                r = hashmap_ensure_put(&network->stacked_netdevs, &string_hash_ops, netdev->ifname, netdev);
+                if (r == -ENOMEM)
                         return log_oom();
-
-                r = hashmap_put(network->stacked_netdevs, netdev->ifname, netdev);
-                if (r < 0)
+                else if(r < 0)
```
Comment (bluca) [inline src/network/networkd-network.c]: same here

File: src/network/networkd-network.c [hunk 3]
```diff
@@ -142,12 +142,10 @@ static int network_resolve_stacked_netdevs(Network *network) {
...
 
-                r = hashmap_ensure_allocated(&network->stacked_netdevs, &string_hash_ops);
-                if (r < 0)
+                r = hashmap_ensure_put(&network->stacked_netdevs, &string_hash_ops, netdev->ifname, netdev);
+                if (r == -ENOMEM)
                         return log_oom();
-
-                r = hashmap_put(network->stacked_netdevs, netdev->ifname, netdev);
-                if (r < 0)
+                if(r < 0)
```
Comment (yuwata) [inline src/network/networkd-network.c]: Missing white space after `if`.

**Review by yuwata — CHANGES_REQUESTED** [review]
One more nitpicky request.

**Comment by ssahani** [comment]
Updated thanks for the review ~

**Comment by yuwata** [comment]
LGTM.

**Pull request merged**